### PR TITLE
Feature/sc 5675 login page and start page to new structure

### PR DESCRIPTION
**Pull request by kgorner**
# Description
within the refactoring, we need to move selectors and methods from one file to another

## Links to Tickets or other pull requests
https://ticketsystem.schul-cloud.org/browse/SC-5675

## Changes
selectors and methods of 2 files moved into the new pageObject structure
usage of logout and login steps that are for now in commonKGO-steps

## Approval for review
- [ ] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### Link to Definiton of Done
More and detailed information on the *definition of done* can be found [on Confluence](https://docs.schul-cloud.org/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

File: page-objects/homework.js [hunk 1]
```diff
@@ -255,12 +259,12 @@ module.exports = {
 
 	submitHomework: async function (taskName, student) {
 		// 	login as student
-		await this.studentLogsIn(student.login, student.password);
+		//await this.studentLogsIn(student.login, student.password);
```
Comment (d3n153) [inline page-objects/homework.js:262]: Can we delete this?

File: shared-objects/loginData.js [hunk 2]
```diff
@@ -15,7 +15,7 @@ module.exports = {
   defaultAdminPassword: 'Schulcloud1!',
 
   elem: {
-    frontpageLoginBtn: 'a[data-testid="login-btn"]',
+    //frontpageLoginBtn: 'a[data-testid="login-btn"]',
```
Comment (d3n153) [inline shared-objects/loginData.js:18]: Can we delete this?

File: shared_steps/firstLogin.js [hunk 3]
```diff
@@ -94,16 +95,18 @@ module.exports = {
 		return name;
 	},
 	logout: async function() {
-		let icon = await driver.$('[data-testid="initials"]');
-		await icon.click();
-		let logOut = await driver.$('[data-testid="logout"]');
-		await logOut.waitForDisplayed(3000);
-		await logOut.click();
-		let frontpageLoginBtn = await driver.$(Login.elem.frontpageLoginBtn);
-		await frontpageLoginBtn.waitForDisplayed(3000);
```
Comment (d3n153) [inline shared_steps/firstLogin.js:103]: At lot of commented out code. Row 98-104, 108.

File: step_definitions/adminLogin-steps.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 'use strict';
 const adminLogin = require('../page-objects/adminLogin');
-const loginPage = require('../page-objects/pages/loginPage');
-const startPage = require('../page-objects/pages/startPage');
+const loginPage = require('../page-objects/pages/generalPagesBeforeLogin/LoginPage.js');
+const startPage = require('../page-objects//pages/generalPagesBeforeLogin/StartPageBeforeLogin.js');
```
Comment (d3n153) [inline step_definitions/adminLogin-steps.js:4]: A / to much before pages?

File: step_definitions/administartion-steps.js [hunk 5]
```diff
@@ -4,46 +4,45 @@ let addPupilToTheCourse = require('../page-objects/addPupilToTheCourse');
 let administration = require('../page-objects/administration');
 //let teacherLogin = require('../page-objects/teacherLogin');
 const elementHelpers = require('../runtime/helpers/elementHelpers.js')
-const loginPage = require('../page-objects/pages/loginPage');
-const startPage = require('../page-objects/pages/startPage');
+const loginPage = require('../page-objects/pages/generalPagesBeforeLogin/LoginPage.js');
+const startPage = require('../page-objects//pages/generalPagesBeforeLogin/StartPageBeforeLogin.js');
```
Comment (d3n153) [inline step_definitions/administartion-steps.js:8]: See comment above.

File: step_definitions/copyCourse-steps.js [hunk 6]
```diff
@@ -1,7 +1,7 @@
 'use strict';
-const loginPage = require('../page-objects/pages/loginPage');
+const loginPage = require('../page-objects/pages/generalPagesBeforeLogin/LoginPage.js');
 const createCourse = require('../page-objects/createCourse');
-const startPage = require('../page-objects/pages/startPage');
+const startPage = require('../page-objects//pages/generalPagesBeforeLogin/StartPageBeforeLogin.js');
```
Comment (d3n153) [inline step_definitions/copyCourse-steps.js:4]: See comment above.

File: step_definitions/createCourse-steps.js [hunk 7]
```diff
@@ -1,8 +1,8 @@
 'use strict';
 
 const createCourse = require('../page-objects/createCourse');
-const loginPage = require('../page-objects/pages/loginPage');
-const startPage = require('../page-objects/pages/startPage');
+const loginPage = require('../page-objects/pages/generalPagesBeforeLogin/LoginPage.js');
+const startPage = require('../page-objects//pages/generalPagesBeforeLogin/StartPageBeforeLogin.js');
```
Comment (d3n153) [inline step_definitions/createCourse-steps.js:5]: See comment above.

File: step_definitions/createTask-steps.js [hunk 8]
```diff
@@ -3,8 +3,8 @@
 const createTask = require("../page-objects/createTask");
 const loginData = require("../shared-objects/loginData");
 const firstLogin = require("../shared_steps/firstLogin.js");
-const loginPage = require('../page-objects/pages/loginPage');
-const startPage = require('../page-objects/pages/startPage');
+const loginPage = require('../page-objects/pages/generalPagesBeforeLogin/LoginPage.js');
+const startPage = require('../page-objects//pages/generalPagesBeforeLogin/StartPageBeforeLogin.js');
```
Comment (d3n153) [inline step_definitions/createTask-steps.js:7]: See comment above.

File: step_definitions/createTeam-steps.js [hunk 9]
```diff
@@ -1,7 +1,7 @@
 'use strict';
 
-const loginPage = require('../page-objects/pages/loginPage');
-const startPage = require('../page-objects/pages/startPage');
+const loginPage = require('../page-objects/pages/generalPagesBeforeLogin/LoginPage.js');
+const startPage = require('../page-objects//pages/generalPagesBeforeLogin/StartPageBeforeLogin.js');
```
Comment (d3n153) [inline step_definitions/createTeam-steps.js:4]: See comment above.

File: step_definitions/homework-steps.js [hunk 10]
```diff
@@ -1,8 +1,8 @@
 'use strict';
 const path = require('path');
 
-const loginPage = require('../page-objects/pages/loginPage');
-const startPage = require('../page-objects/pages/startPage');
+const loginPage = require('../page-objects/pages/generalPagesBeforeLogin/LoginPage.js');
+const startPage = require('../page-objects//pages/generalPagesBeforeLogin/StartPageBeforeLogin.js');
```
Comment (d3n153) [inline step_definitions/homework-steps.js:5]: See comment above.

File: step_definitions/news-steps.js [hunk 11]
```diff
@@ -1,6 +1,6 @@
 'use strict';
-const loginPage = require('../page-objects/pages/loginPage');
-const startPage = require('../page-objects/pages/startPage');
+const loginPage = require('../page-objects/pages/generalPagesBeforeLogin/LoginPage.js');
+const startPage = require('../page-objects//pages/generalPagesBeforeLogin/StartPageBeforeLogin.js');
```
Comment (d3n153) [inline step_definitions/news-steps.js:3]: See comment above.

File: step_definitions/pupilLogin-steps.js [hunk 12]
```diff
@@ -1,24 +1,24 @@
 'use strict';
 
 const elementHelpers = require('../runtime/helpers/elementHelpers.js');
-const loginPage = require('../page-objects/pages/loginPage');
-const startPage = require('../page-objects/pages/startPage');
+const loginPage = require('../page-objects/pages/generalPagesBeforeLogin/LoginPage.js');
+const startPage = require('../page-objects//pages/generalPagesBeforeLogin/StartPageBeforeLogin.js');
```
Comment (d3n153) [inline step_definitions/pupilLogin-steps.js:5]: See comment above.

File: step_definitions/searchCourse-steps.js [hunk 13]
```diff
@@ -1,5 +1,5 @@
-const loginPage = require('../page-objects/pages/loginPage');
-const startPage = require('../page-objects/pages/startPage');
+const loginPage = require('../page-objects/pages/generalPagesBeforeLogin/LoginPage.js');
+const startPage = require('../page-objects//pages/generalPagesBeforeLogin/StartPageBeforeLogin.js');
```
Comment (d3n153) [inline step_definitions/searchCourse-steps.js:2]: See comment above.

File: step_definitions/teacherLogin-steps.js [hunk 14]
```diff
@@ -1,6 +1,6 @@
 'use strict';
-const loginPage = require('../page-objects/pages/loginPage');
-const startPage = require('../page-objects/pages/startPage');
+const loginPage = require('../page-objects/pages/generalPagesBeforeLogin/LoginPage.js');
+const startPage = require('../page-objects//pages/generalPagesBeforeLogin/StartPageBeforeLogin.js');
```
Comment (d3n153) [inline step_definitions/teacherLogin-steps.js:3]: See comment above.

File: step_definitions/visibilityOfAdministrationTabsForAdmin-steps.js [hunk 15]
```diff
@@ -1,6 +1,6 @@
 'use strict';
-const loginPage = require('../page-objects/pages/loginPage');
-const startPage = require('../page-objects/pages/startPage');
+const loginPage = require('../page-objects/pages/generalPagesBeforeLogin/LoginPage.js');
+const startPage = require('../page-objects//pages/generalPagesBeforeLogin/StartPageBeforeLogin.js');
```
Comment (d3n153) [inline step_definitions/visibilityOfAdministrationTabsForAdmin-steps.js:3]: See comment above.

File: step_definitions/visibilityOfDialoguesTabsForTheTeacher-steps.js [hunk 16]
```diff
@@ -1,6 +1,6 @@
 'use strict';
-const loginPage = require('../page-objects/pages/loginPage');
-const startPage = require('../page-objects/pages/startPage');
+const loginPage = require('../page-objects/pages/generalPagesBeforeLogin/LoginPage.js');
+const startPage = require('../page-objects//pages/generalPagesBeforeLogin/StartPageBeforeLogin.js');
```
Comment (d3n153) [inline step_definitions/visibilityOfDialoguesTabsForTheTeacher-steps.js:3]: See comment above.

File: step_definitions/wrongPasswordLogin-steps.js [hunk 17]
```diff
@@ -1,8 +1,8 @@
 'use strict';
 
 const elementHelpers = require('../runtime/helpers/elementHelpers.js');
-const loginPage = require('../page-objects/pages/loginPage');
-const startPage = require('../page-objects/pages/startPage');
+const loginPage = require('../page-objects/pages/generalPagesBeforeLogin/LoginPage.js');
+const startPage = require('../page-objects//pages/generalPagesBeforeLogin/StartPageBeforeLogin.js');
```
Comment (d3n153) [inline step_definitions/wrongPasswordLogin-steps.js:5]: See comment above.

File: page-objects/homework.js [hunk 18]
```diff
@@ -138,15 +141,16 @@ module.exports = {
 
 	// other user logs in to verify
 	studentLogsIn: async function (username, password) {
-		await this.userLogsOut();
-		await firstLogin.pupilLogin(username, password);
-		await firstLogin.firstLoginPupilFullAge(username, password);
+		// await this.userLogsOut();
```
[the remark this answers is not in the document]
Comment (kgorner) [reply page-objects/homework.js:144]: deleted and some the unused methods as well

File: page-objects/homework.js (hunk 1 above)
Comment (kgorner) [reply page-objects/homework.js:262]: deleted

File: shared-objects/loginData.js (hunk 2 above)
Comment (kgorner) [reply shared-objects/loginData.js:18]: deleted

File: step_definitions/adminLogin-steps.js (hunk 4 above)
Comment (kgorner) [reply step_definitions/adminLogin-steps.js:4]: done

File: step_definitions/searchCourse-steps.js (hunk 13 above)
Comment (kgorner) [reply step_definitions/searchCourse-steps.js:2]: done

File: step_definitions/teacherLogin-steps.js (hunk 14 above)
Comment (kgorner) [reply step_definitions/teacherLogin-steps.js:3]: done

File: step_definitions/visibilityOfAdministrationTabsForAdmin-steps.js (hunk 15 above)
Comment (kgorner) [reply step_definitions/visibilityOfAdministrationTabsForAdmin-steps.js:3]: done

File: step_definitions/visibilityOfDialoguesTabsForTheTeacher-steps.js (hunk 16 above)
Comment (kgorner) [reply step_definitions/visibilityOfDialoguesTabsForTheTeacher-steps.js:3]: done

File: step_definitions/wrongPasswordLogin-steps.js (hunk 17 above)
Comment (kgorner) [reply step_definitions/wrongPasswordLogin-steps.js:5]: done

**Comment by kgorner** [comment]
PR (scenario @adminCreateClassWithoutParticipants)
is blocked by bug: https://ticketsystem.schul-cloud.org/browse/SC-5883

File: shared_steps/firstLogin.js (hunk 3 above)
Comment (kgorner) [reply shared_steps/firstLogin.js:103]: done

File: features/administration.feature [hunk 19]
```diff
@@ -11,9 +11,13 @@ Feature: Administrate pupils, classes and teachers
 		When an admin puts in <firstName> and <secondName> and <email> of the new pupil
 		Then the admin should see new pupil with email <email> among his pupils
 		Then the teacher can manually submit a consent <email>
-		Then new pupil can log in
-		Then new pupil accepts data protection policy and sets new password for the profile
+		And log out
+		And go from start page to login page
+		And log in with <username> and <password>
```
Comment (alstanki) [inline features/administration.feature:16]: who is log in?

Comment (kgorner) [reply features/administration.feature:16]: is now changed to:
When admin logs in with email <adminsUsername> and password <password>
		Then admin goes to administration
		And an admin puts in <firstName> and <secondName> and <email> of the new pupil
		And the admin should see new pupil with email <email> among his pupils
		And admin manually submits a consent <email>
		And admin logs out

File: page-objects/homework.js [hunk 20]
```diff
@@ -97,12 +100,12 @@ module.exports = {
 			const containerWithTasks = await driver.$('.col-xl-12');
 			await containerWithTasks.waitForExist(2000);
 			let numOfElems = await containerWithTasks.$$('li');
```
Comment (alstanki) [inline page-objects/homework.js:102]:
```suggestion
			let numOfElems = (await containerWithTasks.$$('li')).length;
```

File: page-objects/homework.js [hunk 21]
```diff
@@ -97,12 +100,12 @@ module.exports = {
...
-			for (var i=1; i<=numOfElems.length-1; i++) {
-					let nameOfTheTaskSelector = await driver.$('.col-xl-12 > li:nth-child('+i+') > .content > h2' );
-					let nameOfTheTask = await nameOfTheTaskSelector.getText();
-					if(await nameOfTheTask.includes(taskname)) {
-						return i;
-					}
+			for (var i = 1; i <= numOfElems.length - 1; i++) {
```
Comment (alstanki) [inline page-objects/homework.js:103]:
```suggestion
			for (var i = 1; i <= numOfElems; i++) {
```
This part may be problematic in original form. As it skips last element.

File: page-objects/homework.js [hunk 22]
```diff
@@ -97,12 +100,12 @@ module.exports = {
...
+				let nameOfTheTaskSelector = await driver.$('.col-xl-12 > li:nth-child(' + i + ') > .content > h2');
+				let nameOfTheTask = await nameOfTheTaskSelector.getText();
+				if (await nameOfTheTask.includes(taskname)) {
+					return i;
+				}
 			}
 		}
 		return;
```
Comment (alstanki) [inline page-objects/homework.js:111]: Return what? -> undefined will be returned

File: features/administration.feature [hunk 23]
```diff
@@ -2,18 +2,20 @@
...
 	Scenario Outline: Admin creates a pupil
-		When admin goes to administration
-		When an admin puts in <firstName> and <secondName> and <email> of the new pupil
-		Then the admin should see new pupil with email <email> among his pupils
-		Then the teacher can manually submit a consent <email>
-		Then new pupil can log in
-		Then new pupil accepts data protection policy and sets new password for the profile
+		When admin logs in with email <adminsUsername> and password <password>
+		Then admin goes to administration
+		And an admin puts in <firstName> and <secondName> and <email> of the new pupil
```
[the remark this answers is not in the document]
Comment (kgorner) [reply features/administration.feature:12]: Das mache ich gerne, wenn wir tatsächlich die Tests aufräumen. Hier ging es erstmal um das transferieren von einer in die andere Datei.

[the remark this answers is not in the document]
Comment (kgorner) [reply features/administration.feature:12]: Das mache ich gerne, wenn wir tatsächlich die Tests aufräumen. Hier ging es erstmal um das transferieren von einer in die andere Datei.

File: features/editProfile.feature [hunk 24]
```diff
@@ -2,21 +2,19 @@
...
 	Background:
-		Given user arrives on the Schul-Cloud homepage
+		Given teacher arrives on the Schul-Cloud homepage
 
 	@teacherChangesPassword
 	Scenario Outline: teacher changes the passwort
-		When go from start page to login page
-		And log in with <username> and <password>
+		When teacher logs in with email <username> and password <password>
 		And the teacher should accept the data protection
```
[the remark this answers is not in the document]
Comment (kgorner) [reply features/editProfile.feature:10]: Das mache ich gerne, wenn wir tatsächlich die Tests aufräumen. Hier ging es erstmal um das transferieren von einer in die andere Datei.

File: features/homework.feature [hunk 25]
```diff
@@ -2,52 +2,55 @@
 Feature: create different types of homework
 
 
-Background: a teacher logs in and creates a course
-Given the teacher starts on the login page
-Given the teacher is logged-in successfully
-Given the teacher goes to the course page as a next step
+    Background: a teacher logs in and creates a course
+        Given the teacher starts on the login page
```
[the remark this answers is not in the document]
Comment (kgorner) [reply features/homework.feature:6]: Das mache ich gerne, wenn wir tatsächlich die Tests aufräumen. Hier ging es erstmal um das transferieren von einer in die andere Datei.

File: features/navigationLeft.feature [hunk 26]
```diff
@@ -3,7 +3,7 @@ Feature:checkLeftNavigation
 
 	@teacherclicksLeftMenuItems
 	Scenario Outline: The user logs in as a teacher and clicks the left menu items he should have access to
-		When The Teacher arrives on the Schul-Cloud page
+		When teacher arrives on the Schul-Cloud homepage
 		When a teacher logs in his account using <username> and <password>
```
[the remark this answers is not in the document]
Comment (kgorner) [reply features/navigationLeft.feature:7]: is done with the last commits

File: page-objects/homework.js [hunk 27]
```diff
@@ -136,17 +139,10 @@ module.exports = {
...
-		await this.userLogsOut();
-		await firstLogin.pupilLogin(username, password);
-		await firstLogin.firstLoginPupilFullAge(username, password);
-	},
 	teacherLogsIn: async function () {
 		await this.userLogsOut();
-		let frontpageLoginBtn = await driver.$(Login.elem.frontpageLoginBtn);
-		await frontpageLoginBtn.click();
-		await loginPage.performLogin(Login.defaultTeacherUsername,Login.defaultTeacherpassword);
+		await startPage.clickLoginBtn();
```
[the remark this answers is not in the document]
Comment (kgorner) [reply page-objects/homework.js:144]: ja, habe eine Methode geschrieben, somit können wir die von überall nutzen und müssen nicht überall wieder Selektoren duplizieren. Ich erkläre dir das heute mal.

File: step_definitions/createCourse-steps.js [hunk 28]
```diff
@@ -1,13 +1,14 @@
 const elementHelpers = require("../runtime/helpers/elementHelpers.js");
 const Login = require("../shared-objects/loginData");
-const loginPage = require("../page-objects/pages/loginPage");
-const startPage = require("../page-objects/pages/startPage");
+//const loginPage = require("../page-objects/pages/loginPage");
```
[the remark this answers is not in the document]
Comment (kgorner) [reply step_definitions/createCourse-steps.js:3]: hab's entfernt

File: features/administration.feature [hunk 29]
```diff
@@ -11,9 +11,13 @@ Feature: Administrate pupils, classes and teachers
 		When an admin puts in <firstName> and <secondName> and <email> of the new pupil
 		Then the admin should see new pupil with email <email> among his pupils
 		Then the teacher can manually submit a consent <email>
-		Then new pupil can log in
-		Then new pupil accepts data protection policy and sets new password for the profile
+		And log out
+		And go from start page to login page
```
[the remark this answers is not in the document]
Comment (kgorner) [reply features/administration.feature]: noButton
Button
as you can see the button is only on startpage, but I changed it anyways to:
`Background:
		Given admin arrives on the Schul-Cloud homepage

	Scenario Outline: Admin creates a pupil
		When admin logs in with email <adminsUsername> and password <password>
		Then admin goes to administration
		And an admin puts in <firstName> and <secondName> and <email> of the new pupil
		And the admin should see new pupil with email <email> among his pupils
		And admin manually submits a consent <email>
		And admin logs out
		#Then new pupil can log in
		#Then new pupil accepts data protection policy and sets new password for the profile`
and opened a ticket to repair this test scenario: https://ticketsystem.schul-cloud.org/browse/TTT-267

File: features/administration.feature [hunk 30]
```diff
@@ -11,9 +11,13 @@ Feature: Administrate pupils, classes and teachers
...
+		And log in with <username> and <password>
+		And the pupil should accept the data protection
```
[the remark this answers is not in the document]
Comment (kgorner) [reply features/administration.feature]: will be done later - this was a ticket of transferring, not of improving the existing scenarios

File: features/administration.feature [hunk 31]
```diff
@@ -11,9 +11,13 @@ Feature: Administrate pupils, classes and teachers
...
+		#Then new pupil can log in
+		#Then new pupil accepts data protection policy and sets new password for the profile
 
 		Examples:
-			|firstName|secondName|email|
-			|Georg|Georgmann|georgmann@schul-cloud.org|
+			| firstName | secondName | email                     | username                    | password     |
```
[the remark this answers is not in the document]
Comment (kgorner) [reply features/administration.feature]: things like that will be done later - this was a ticket of transferring, not of improving the existing scenarios
now it is like that: 
`Examples:
			| firstName | secondName | email                     | adminsUsername        | password     |
			| Georg     | Georgmann  | georgmann@schul-cloud.org | admin@schul-cloud.org | Schulcloud1! |`

File: shared_steps/firstLogin.js [hunk 32]
```diff
@@ -94,16 +95,10 @@ module.exports = {
 		return name;
```
[the remark this answers is not in the document]
Comment (kgorner) [reply shared_steps/firstLogin.js:96]: This is something I didn't change. It just did a transfer-ticket.
in this file I can only find "return name" in the following method - and there it is definded:
`getNameAndPosition: async function() {
		let userIcon = await driver.$('.btn-avatar > a');
		await userIcon.click();

		let nameBox = await driver.$(
			'.dropdown-name'
		);
		let name = await nameBox.getText();
		return name;
	},`

File: step_definitions/pupilLogin-steps.js [hunk 33]
```diff
@@ -1,24 +1,23 @@
...
 const elementHelpers = require('../runtime/helpers/elementHelpers.js');
-const loginPage = require('../page-objects/pages/loginPage');
-const startPage = require('../page-objects/pages/startPage');
+const loginPage = require('../page-objects/pages/generalPagesBeforeLogin/LoginPage.js');
+const startPage = require('../page-objects/pages/generalPagesBeforeLogin/StartPageBeforeLogin.js');
 const loginData = require('../shared-objects/loginData');
-
 const firstLogin = require('../shared_steps/firstLogin.js');
 
 Given(/^a pupil arrives on the Schul-Cloud login homepage$/, function () {
```
[the remark this answers is not in the document]
Comment (kgorner) [reply step_definitions/pupilLogin-steps.js]: Such things will be done later in improvement of existing stuff. This was just about transferring to new page object structure

File: page-objects/homework.js (hunk 21 above)
Comment (kgorner) [reply page-objects/homework.js:103]: done with next commit

File: page-objects/homework.js (hunk 22 above)
Comment (kgorner) [reply page-objects/homework.js:111]: done with next commit

**Pull request merged**